lua/rainbow/internal.lua

```diff
@@ -34,54 +34,44 @@ local function color_no(mynode, len)
         end
 end
 
-local callbackfn = function(bufnr, parser)
-        -- no need to do anything when pum is open
-        if vim.fn.pumvisible() == 1 then
-                return
-        end
+local callbackfn = function(bufnr, _, changes, tree, lang)
+        --vim.schedule_wrap(function()
```

Comment:
maybe one could experiment to do this async `schedule_wrap` alone won't make it async